flume-ng-core/src/test/java/org/apache/flume/source/TestSyslogUtils.java

```diff
@@ -196,41 +199,53 @@ public void TestRfc3164HeaderApacheLogWithNulls() throws ParseException {
    */
   @Test
   public void TestRfc3164Dates() throws ParseException {
-    for (int i = -10; i <= 1; i++) {
-      SimpleDateFormat sdf = new SimpleDateFormat("MMM  d hh:MM:ss");
-      Date date = new Date(System.currentTimeMillis());
-      Calendar cal = Calendar.getInstance();
-      cal.setTime(date);
-      cal.add(Calendar.MONTH, i);
-
-      //Small tweak to avoid the 1 month in the future ticking over by a few seconds between now
-      //and when the checkHeader actually runs
-      if (i == 1) {
-        cal.add(Calendar.DAY_OF_MONTH, -1);
+    //We're going to run this test using a mocked clock, once for the next 13 months
+    for (int monthOffset = 0; monthOffset <= 13; monthOffset++) {
+      Clock mockClock = Clock.fixed(
+              LocalDateTime.now().plusMonths(monthOffset).toInstant(ZoneOffset.UTC),
+              Clock.systemDefaultZone().getZone()
+      );
+
+      //We're then going to try input dates (without the year) for all 12 months, starting
+      //10 months ago, and finishing next month (all relative to our mocked clock)
+      for (int i = -10; i <= 1; i++) {
+        SimpleDateFormat sdf = new SimpleDateFormat("MMM  d hh:MM:ss", Locale.ENGLISH);
```

Comment:
@szaboferee @mcsanady - I think the date format should be changed to `MMM dd hh:mm:ss`? I left a comment for @tmgstevens in my PR but haven't heard back from him yet.

Reply:
@vegiapparao  good point.  Now minutes are referenced as months in the pattern. Luckily it is in  a test, so no production is affected. We should address that in a following pull request backed up with a jira ticket:.

Reply:
Agreed, although I'm now confused as to why the test passes.

Reply:
@tmgstevens, @szaboferee,  I have changed it locally, ran the full suite and all the tests are passing.  Shall I submit a patch for this?

Reply:
@vegiapparao okay, I've finally got my head around this...! SimpleDateFormat sdf is only used to format the desired date for the message. The desired date is then tested against the same date string with yyyy prepended, to make sure the the year has been inferred correctly. As such, the minute actually doesn't matter. It's useful to tidy up, but has no functional impact. Up to you if you want to raise a JIRA and submit a PR.